ui/apps/platform/src/Containers/SystemConfig/configUtils.ts

```diff
@@ -0,0 +1,51 @@
+import { PlatformComponentRule, PlatformComponentsConfig } from 'types/config.proto';
+
+export type PlatformComponentsConfigRules = {
+    coreSystemRule: PlatformComponentRule;
+    redHatLayeredProductsRule: PlatformComponentRule;
+    customRules: PlatformComponentRule[];
+};
```

Comment:
Encapsulating the flat nature of the API response in our own purpose-built type is a good call here. 👍